pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java

```diff
@@ -148,6 +156,15 @@ public void processSqlQueryPost(String query, @Suspended AsyncResponse asyncResp
       BrokerResponse brokerResponse =
           executeSqlQuery((ObjectNode) requestJson, makeHttpIdentity(requestContext), false);
       asyncResponse.resume(brokerResponse.toJsonString());
+    } catch (ServiceUnavailableException e) {
```

Comment:
Can you confirm the error message returned to the user?

I think you had mentioned that Grizzly was returning the error above the handler code so the error message being returned was "Internal error".

Reply:
Thanks Ankit, I just checked again. The ServiceUnavailableException is not getting propagated to the API and it is caught by the Grizzly HTTP server and handled internally. Users will receive a error code 503 (Service Unavailable).

I will fix it and propagate the exception to the API.

Reply:
Addressed.